airflow/providers/amazon/aws/hooks/ecs.py

```diff
@@ -198,7 +196,18 @@ def _event_to_str(self, event: dict) -> str:
         return f"[{formatted_event_dt}] {message}"
 
     def get_last_log_messages(self, number_messages) -> list:
-        return [log["message"] for log in deque(self._get_log_events(), maxlen=number_messages)]
+        """
+        Gets the last logs messages in one single request, so restrictions apply:
+         - if logs are too old, the response will be empty
```

Comment:
this may look bad, but from some tries I did, it'd require many calls to get logs from the tail when they are old (logs that are 1 week old already require tens of calls before one will return something non-empty), and the existing code was giving up after 3 empty responses, so we're only making the problem appear a little faster.
And from the ecs hook, we can assume it's going to used to query fresh logs from tasks that just finished.

Reply:
I'm not sure I grok this. Why should reading from the bottom of a log stream be any different depending on how old the stream is?

Reply:
well that's a question for cloudwatch teams, but somehow the older the last event in the log stream, the more empty requests you have to pull before you start getting results.
I guess maybe they try to answer within a certain time SLA ? And that doesn't give them enough time to get it from semi-hot storage ? Or like they start searching in chronological order, and the older it is, the more records they need to go through before getting hits ?